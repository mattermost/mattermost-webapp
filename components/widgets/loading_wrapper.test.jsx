// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import LoadingWrapper from './loading_wrapper.jsx';

describe('components/widgets/LoadingWrapper', () => {
    const testCases = [
        {
            name: 'showing spinner with text',
            loading: true,
            type: 'spinner',
            text: 'test',
            children: 'children',
        },
        {
            name: 'showing spinner without text',
            loading: true,
            type: 'spinner',
            children: 'text',
        },
        {
            name: 'showing bars with text',
            loading: true,
            type: 'bars',
            text: 'test',
            children: 'text',
        },
        {
            name: 'showing bars without text',
            loading: true,
            type: 'bars',
            children: 'text',
        },
        {
            name: 'showing content with children',
            loading: false,
            children: 'text',
        },
        {
            name: 'showing content without children',
            loading: false,
        },
    ];
    for (const testCase of testCases) {
        test(testCase.name, () => {
            const wrapper = shallow(
                <LoadingWrapper
                    loading={testCase.loading}
                    text={testCase.text}
                    type={testCase.type}
                >
                    {testCase.children}
                </LoadingWrapper>
            );
            expect(wrapper).toMatchSnapshot();
        });
    }
});
