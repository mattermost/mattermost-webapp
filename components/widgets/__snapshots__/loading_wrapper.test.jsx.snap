// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/widgets/LoadingWrapper showing bars with text 1`] = `
<span
  className="loading-component with-text"
>
  <img
    className="spinner"
    src={null}
  />
  test
</span>
`;

exports[`components/widgets/LoadingWrapper showing bars without text 1`] = `
<span
  className="loading-component"
>
  <img
    className="spinner"
    src={null}
  />
</span>
`;

exports[`components/widgets/LoadingWrapper showing content with children 1`] = `"text"`;

exports[`components/widgets/LoadingWrapper showing content without children 1`] = `""`;

exports[`components/widgets/LoadingWrapper showing spinner with text 1`] = `
<span
  className="loading-component with-text"
>
  <span
    className="fa fa-spinner fa-pulse spinner"
    title="Loading Icon"
  />
  test
</span>
`;

exports[`components/widgets/LoadingWrapper showing spinner without text 1`] = `
<span
  className="loading-component"
>
  <span
    className="fa fa-spinner fa-pulse spinner"
    title="Loading Icon"
  />
</span>
`;
